src/cascadia/TerminalSettingsEditor/ColorSchemesPageViewModel.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "ColorSchemesPageViewModel.h"
+#include "ColorSchemesPageViewModel.g.cpp"
+
+#include <LibraryResources.h>
+#include "..\WinRTUtils\inc\Utils.h"
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    ColorSchemesPageViewModel::ColorSchemesPageViewModel(const Model::CascadiaSettings& settings) :
+        _settings{ settings }
+    {
+        _MakeColorSchemeVMsHelper();
+        CurrentScheme(_AllColorSchemes.GetAt(0));
+    }
+
+    void ColorSchemesPageViewModel::UpdateSettings(const Model::CascadiaSettings& settings)
+    {
+        _settings = settings;
+        _MakeColorSchemeVMsHelper();
+    }
+
+    void ColorSchemesPageViewModel::RequestSetCurrentScheme(Editor::ColorSchemeViewModel scheme)
+    {
+        CurrentScheme(scheme);
+
+        // Update the state of the page
+        _PropertyChangedHandlers(*this, Windows::UI::Xaml::Data::PropertyChangedEventArgs{ L"CanDeleteCurrentScheme" });
+    }

```

Comment:
When `CurrentScheme` is called we already send a property changed event (its baked into the `WINRT_OBSERVABLE_PROPERTY` macro, which we used to initialize the `CurrentScheme` property), so everything in the xaml that cares about `CurrentScheme` like `CurrentScheme.ColorEntryAt(0)` or `CurrentScheme.CurrentForegroundColor` automatically update. Since `CanDeleteCurrentScheme` is not a property on `CurrentScheme`, we need to send a separate event

Reply:
> 2. When `CurrentScheme` is called...

Oh right. That's that weird thing where binding via markup works as expected, but binding via code does not. Ugh.

Reply:
Ah, yes. This is where a multilevel binding like `CurrentColorScheme.CanDelete` would work properly (xaml will listen to both properties changing.) The issue arises when you split up CanDelete between the CSPVM and the CSVM...

Reply:
as for 1... do you have a concrete idea for ones that it would reject? it's okay to design for what we have today, not what we'll need in the unknowable future :)